File: java/Pagination/app/src/main/java/com/singularitycoder/pagination/ApiEndPoints.java
package com.singularitycoder.pagination;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiEndPoints {

    @GET("/api/users")
    Call<UsersResponse> getUsersList(
            @Query("page") String pageNumber,
            @Query("per_page") String numOfItemsPerPage
    );
}

File: kotlin/KotlinRetrofit1/app/src/main/java/com/singularitycoder/kotlinretrofit1/helper/api/ApiEndPoints.kt
package com.singularitycoder.kotlinretrofit1.helper.api

import com.singularitycoder.kotlinretrofit1.model.RepoResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiEndPoints {

    @GET("/search/repositories")
    fun searchAllRepositories(
            @Query("q") query: String,
            @Query("sort") sort: String,
            @Query("order") order: String,
    ): Call<RepoResponse>
}